### PR TITLE
Improved error logging when adding rules

### PR DESCRIPTION
**Pull request by bergvandenp**
I stumbled upon the same issue as described here:

https://community.openhab.org/t/solved-oh3-issue-unable-to-create-rules-scripts-schedules/111325
https://community.openhab.org/t/openhab-3-error-creating-rules-or-scripts-using-blocky/111422

This makes the logging a bit more useful

### CODE REVIEW

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
We do not log stacktraces when there is no bug in openHAB itself.
What can be done is to log exceptions when debugging is enabled for a given class.

**Comment by NorbertHD** [comment]
@bergvandenp can you please change it to:
````
            if (logger.isDebugEnabled()) {
                logger.warn(errMessage, e);
            } else {
                logger.warn("{}", errMessage);
            }
````

**Review by kaikreuzer — APPROVED** [review]
thanks!

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Now you only have to make SAT happy:
```
[ERROR] Code Analysis Tool has found: 
 1 error(s)! 
 4 warning(s) 
 1 info(s)
[ERROR] org.openhab.core.automation.rest.internal.RuleResource.java:[192]
Format should be constant. Use placeholder to reduce the needless cost of parameter construction. see http://www.slf4j.org/faq.html#logging_performance
```

File: bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java [hunk 1]
```diff
@@ -178,15 +178,23 @@ public Response create(@Parameter(description = "rule data", required = true) Ru
...
             String errMessage = "Creation of the rule is refused: " + e.getMessage();
-            logger.warn("{}", errMessage);
+            logException(e, errMessage);
             return JSONResponse.createErrorResponse(Status.BAD_REQUEST, errMessage);
         }
     }
 
+    private void logException(RuntimeException e, String errMessage) {
+        if (logger.isDebugEnabled()) {
+            logger.warn(errMessage, e);
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java:192]:
```suggestion
            logger.warn("{}", errMessage, e);
```

Comment (NorbertHD) [reply bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java:192]: @kaikreuzer does this work?
AFAIK there is no warn​(String format, Object... arguments, Throwable t)

Comment (kaikreuzer) [reply bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java:192]: Yes, that works.

Comment (bergvandenp) [reply bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java]: whoops. my bad. thanks for the suggestion

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**